File: matlab/lib/+IceSSL/EndpointInfo.m
classdef EndpointInfo < Ice.EndpointInfo
    % TCPEndpointInfo   Summary of TCPEndpointInfo
    %
    % Provides access to a TCP endpoint information.

    % Copyright (c) 2003-present ZeroC, Inc. All rights reserved.

    methods
        function obj = EndpointInfo(type, secure, underlying, timeout, compress)
            if nargin == 0
                underlying = [];
                timeout = 0;
                compress = false;
            end
            obj = obj@Ice.EndpointInfo(type, false, secure, underlying, timeout, compress);
        end
    end
end

File: java/src/IceBT/src/main/java/com/zeroc/IceBT/EndpointInfo.java
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

package com.zeroc.IceBT;

/**
 * Provides access to Bluetooth endpoint information.
 **/
public abstract class EndpointInfo extends com.zeroc.Ice.EndpointInfo
{
    public EndpointInfo()
    {
        super();
        this.addr = "";
        this.uuid = "";
    }

    public EndpointInfo(com.zeroc.Ice.EndpointInfo underlying, int timeout, boolean compress, String addr, String uuid)
    {
        super(underlying, timeout, compress);
        this.addr = addr;
        this.uuid = uuid;
    }

    /**
     * The address configured with the endpoint.
     **/
    public String addr;

    /**
     * The UUID configured with the endpoint.
     **/
    public String uuid;

    public EndpointInfo clone()
    {
        return (EndpointInfo)super.clone();
    }

    /** @hidden */
    public static final long serialVersionUID = 4825481358879091449L;
}
